omod-2.0/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs2_0/PatientAllergyController2_0Test.java

```diff
@@ -0,0 +1,244 @@
+		Assert.assertEquals("unit test", allergy.getVoidReason());
+		Assert.assertEquals(Allergies.UNKNOWN, allergies.getAllergyStatus());
+	}
+	
+	/**
+	 *  Save Allergy with Coded Allergen
+	 */
+	@Test
+	public void shouldSaveAllergyWithCodedAllergen() throws Exception
+	{
+		Allergy allergy = Context.getPatientService().getAllergyByUuid(getUuid());
+		Patient patient = allergy.getPatient();
+		Allergies allergies = Context.getPatientService().getAllergies(patient);
+		Assert.assertEquals(allergies.size(), 4);
+
+		String json = "{\"allergen\": {\"allergenType\":\"DRUG\", \"codedAllergen\": { \"uuid\": \"35d3346a-6769-4d52-823f-b4b234bac3e3\"} }}";
+		SimpleObject newAllergy = deserialize(handle(newPostRequest(getURI(), json)));
+		
+		Assert.assertEquals("DRUG", Util.getByPath(newAllergy, "allergen/allergenType"));
+		Assert.assertEquals("35d3346a-6769-4d52-823f-b4b234bac3e3", Util.getByPath(newAllergy, "allergen/codedAllergen/uuid"));
+		allergies = Context.getPatientService().getAllergies(patient);
+		Assert.assertEquals(allergies.size(), 5);
+	}
+	
+	/**
+	 *  Save Allergy with Non Coded Allergen
+	 */
+	@Test
+	public void shouldSaveAllergyWithNonCodedAllergen() throws Exception
+	{
+		Allergy allergy = Context.getPatientService().getAllergyByUuid(getUuid());
+		Patient patient = allergy.getPatient();
+		Allergies allergies = Context.getPatientService().getAllergies(patient);
+		Assert.assertEquals(allergies.size(), 4);
+		
+		// this is the proper sequence to set nonCodedAllergen
+		String json = "{\"allergen\": {\"allergenType\":\"DRUG\", \"codedAllergen\": { \"uuid\": \"5622AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA\"}, \"nonCodedAllergen\": \"test non coded\" }}";
+		SimpleObject newAllergy = deserialize(handle(newPostRequest(getURI(), json)));
+		
+		Assert.assertEquals("DRUG", Util.getByPath(newAllergy, "allergen/allergenType"));
+		Assert.assertEquals("test non coded", Util.getByPath(newAllergy, "allergen/nonCodedAllergen"));
+		Assert.assertEquals("5622AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA", Util.getByPath(newAllergy, "allergen/codedAllergen/uuid"));
+		allergies = Context.getPatientService().getAllergies(patient);
+		Assert.assertEquals(allergies.size(), 5);
+	}
+
+	/**
+	 *  Set No Known Allergies icorrectly using empty post body
+	 *  achieved with PUT with empty post body when list of allergies is empty
+	 *  throw ResourceDoesNotSupportOperationException because Patient does not exist
+	 */
+	@Test(expected = ResourceDoesNotSupportOperationException.class)
+	public void shouldThrowExceptionWhenSettingNoKnownAllergiesForNonExistingPatientUsingEmptyPostBodyAllergiesNotEmpty() throws Exception
+	{
+		Allergy allergy = Context.getPatientService().getAllergyByUuid(getUuid());
+		Patient patient = allergy.getPatient();
+		Allergies allergies = Context.getPatientService().getAllergies(patient);		
+		Assert.assertEquals(Allergies.SEE_LIST, allergies.getAllergyStatus());
+
+		// atempt to set no known allergies with PUT

```

Comment:
I would also rename the method to: shouldThrowExceptionWhenSettingNoKnownAllergiesForNonExistingPatient